pkgdown 2.0.0 (#548)
tidyverse/tidyverse.org: content/blog/pkgdown-2-0-0/index.Rmd

* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
* [x] Find photo & update yaml metadata
* [x] Create `thumbnail-sq.jpg`; height and width should be equal
* [x] Create `thumbnail-wd.jpg`; width should be >5x height
* [x] `hugodown::use_tidy_thumbnails()`
* [x] Add intro sentence, e.g. the standard tagline for the package
* [x] `usethis::use_tidy_thanks()`
-->
```
We're very chuffed to announce the release of [pkgdown](https://pkgdown.r-lib.org) 2.0.0.
pkgdown is designed to make it quick and easy to build a website for your package.
Install it with:

```{r, eval = FALSE}
install.packages("pkgdown")
```

This is a massive release that brings a major visual refresh and a huge number of improvements.
This release would not have been possible without pkgdown's newest author [Maëlle Salmon](https://masalmon.eu), who was the powerhouse behind many of the improvements in this release, particularly the switch to bootstrap 5, improved customisation, and implementation of local search.

There are way too many changes to describe individually here, so this post will focus on the most important new features:

> -   The new template that uses bootstrap 5.

Reviewer: ```suggestion
-   The new template that uses Bootstrap 5.
```

Reviewer: Also, should Bootstrap be introduced with a footnote or so? "[Collection of ready-made HTML, JS and CSS created at Twitter](https://en.wikipedia.org/wiki/Bootstrap_(front-end_framework))"

Reviewer: with some context e.g. "building blocks for pkgdown templates and assets".

Author: Added a little context

-   The most important new ways to customise your site.
-   Some of the biggest changes to code display.
-   A grab bag of other cool features.

> See the [release notes](https://github.com/r-lib/pkgdown/blob/main/NEWS.md) for a complete list of everything that's changed.

Reviewer: I'd argue it'll better on the pkgdown website :joy:
```suggestion
See the [release notes](https://pkgdown.r-lib.org/news/index.html) for a complete list of everything that's changed.
```

```{r setup}
library(pkgdown)
```

## Bootstrap 5

pkgdown comes with a refreshed template that uses [Bootstrap 5](https://getbootstrap.com/docs/5.1/getting-started/introduction/), an major update over the previous [Bootstrap 3](https://getbootstrap.com/docs/3.4/).
Because this is a major change, you'll need to opt-in by setting the `boostrap` version in your `_pkgdown.yml`:

``` yaml
template:
  bootstrap: 5
```

> (The old bootstrap 3 template is superseded; it will continue to work for some time, but it won't gain any new features and we encourage you to switch to the new template the next time you're working on your package.)

Reviewer: ```suggestion
(The old Bootstrap 3 template is superseded; it will continue to work for some time, but it won't gain any new features and we encourage you to switch to the new template the next time you're working on your package.)
```

The new theme includes:

-   A number of minor improvements to accessibility, including a larger font size, greater use of `aria` attributes, and [accessible syntax highlighting colour scheme](https://apreshill.github.io/rmda11y/arrow.html), designed by Alison Hill.

-   Support for new features like in-line footnotes and [tabsets in articles](https://bookdown.org/yihui/rmarkdown-cookbook/html-tabs.html).

-   Since so many packages have lovely logos, the package logo is now displayed on every page and made even more prominent on the home page.

As an added incentive to upgrade your template to Bootstrap 5, you'll get site search for "free": pkgdown now supports searching with no external dependencies and no set-up.
Learn more in `vignette("search")`.

## Customisation

The new template is also much easier to customise.

> A few of the most important features are noted below to whet your appetite; learn more in `vignette("customise")`.

Reviewer: No autolinking of vignettes?

Author: I wondered about that too; it's because they're only on the dev version, which will be published before this blog post is published (so they should be linked in the final blog post)

> -   You can now easily change the overall visual appearance by picking a bootswatch theme:

Reviewer: ```suggestion
-   You can now easily change the overall visual appearance by picking a Bootswatch theme:
```

    ``` yaml
    template:
      bootstrap: 5
      bootswatch: cyborg
    ```

>     Or by selectively overriding the "bslib" variables used to generate the CSS:

Reviewer: should bslib be introduced somewhere earlier in this section or the one before that?

    ``` yaml
    template:
      bootstrap: 5
      bslib:
        bg: "#202123"
        fg: "#B8BCC2"
        primary: "#306cc9"
        base_font: {google: "Roboto"}
    ```

>     You can also choose a different syntax highlighting theme:

Reviewer: where does one get the full list?

Author: Added a reminder to see the vignette for more details below.

    Activate the translations by setting the `lang` field in `_pkgdown.yaml`, e.g.:

    ``` yaml
    lang: fr
    ```

    pkgdown includes translations for Spanish (es), German (de), French (fr), Portuguese (pt), Turkish (tr) and Chinese (zh_CN).
    A big thanks to my colleagues who provided the initial translations, and to @dieghernan, @rivaquiroga, @jplecavalier who supplied additional improvements.
    If you're interested in adding translations for your language please [file an issue](https://github.com/r-lib/pkgdown/issues) and we'll help you get started.

-   You can add arbitrary HTML to every page with the new `includes` parameter.
    This makes it easy to add analytics to your site, e.g. to use [plausible.io](https://plausible.io):

    ``` yaml
    templates:
      includes:
        in_header: |
          <script defer data-domain="{YOUR DOMAIN}" src="https://plausible.io/js/plausible.js"></script>
    ```

-   The author, sidebar, and footer configuration is much more flexible allowing you to customise individual components while keeping most of the defaults (previously customisation was mostly all or nothing).

>     See `?build_home()` and `?build_site()` for details.

Reviewer: not autolinked either

Author: Ok here it was because I had the `()` at the end

---
output: hugodown::hugo_document

slug: pkgdown-2-0-0
title: pkgdown 2.0.0
date: 2021-11-29
author: Hadley Wickham
description: >

>     usethis 2.0.0 includes a major refresh of the default template (now 

Reviewer: ```suggestion
    pkgdown 2.0.0 includes a major refresh of the default template (now 
```

    to code styling, and much, much, more.
    
photo:
  url: https://unsplash.com/photos/1HIKnKtXEU0
  author: Edgar Soto

# one of: "deep-dive", "learn", "package", "programming", "roundup", or "other"
categories: [package] 
tags: [pkgdown, devtools]
---

```{=html}
<!--
TODO:
* [x] Look over / edit the post's title in the yaml
* [x] Edit (or delete) the description; note this appears in the Twitter card
* [x] Pick category and tags (see existing with `hugodown::tidy_show_meta()`)
* [x] Find photo & update yaml metadata
* [x] Create `thumbnail-sq.jpg`; height and width should be equal
* [x] Create `thumbnail-wd.jpg`; width should be >5x height
* [x] `hugodown::use_tidy_thumbnails()`
* [x] Add intro sentence, e.g. the standard tagline for the package
* [x] `usethis::use_tidy_thanks()`
-->
```
We're very chuffed to announce the release of [pkgdown](https://pkgdown.r-lib.org) 2.0.0.
pkgdown is designed to make it quick and easy to build a website for your package.
Install it with:

```{r, eval = FALSE}
install.packages("pkgdown")
```

This is a massive release that brings a major visual refresh and a huge number of improvements.

> This release would not have been possible without pkgdown's newest author [Maëlle Salmon](https://masalmon.eu), who was the powerhouse behind many of the improvements in this release, particularly the switch to bootstrap 5, improved customisation, and implementation of local search.

Reviewer 2: ```suggestion
This release would not have been possible without pkgdown's newest author, [Maëlle Salmon](https://masalmon.eu), who was the powerhouse behind many of the improvements in this release, particularly the switch to Bootstrap 5, improved customisation, and implementation of local search.
```

Reviewer 2: Also, we have bootstrap 5 here and Bootstrap 5 below, so it should just be one or the other. (I went uppercase above)

There are way too many changes to describe individually here, so this post will focus on the most important new features:

-   The new template that uses Bootstrap 5.

> -   The most important new ways to customise your site.

Reviewer 2: Nested "most important"

-   Some of the biggest changes to code display.
-   A grab bag of other cool features.

See the [release notes](https://pkgdown.r-lib.org/news/index.html) for a complete list of everything that's changed.

```{r setup}
library(pkgdown)
```

## Bootstrap 5

> pkgdown comes with a refreshed template that uses [Bootstrap 5](https://getbootstrap.com/docs/5.1/getting-started/introduction/), an major update over the previous [Bootstrap 3](https://getbootstrap.com/docs/3.4/).

Reviewer 2: ```suggestion
pkgdown comes with a refreshed template that uses [Bootstrap 5](https://getbootstrap.com/docs/5.1/getting-started/introduction/), a major update over the previous [Bootstrap 3](https://getbootstrap.com/docs/3.4/).
```

(Bootstrap is the collection of HTML, CSS, and JS files that give pkgdown sites their basic style).
Because this is a major change, you'll need to opt-in by setting the `boostrap` version in your `_pkgdown.yml`:

``` yaml
template:
  bootstrap: 5
```

> (The old Bootstrap 3 template is superseded; it will continue to work for some time, but it won't gain any new features and we encourage you to switch to the new template the next time you're working on your package.)

Reviewer 2: ```suggestion
The old Bootstrap 3 template is superseded; it will continue to work for some time, but it won't gain any new features and we encourage you to switch to the new template the next time you're working on your package.
```

Reviewer 2: I get why this is a parenthetical. But, given it's its own paragraph, I think you can drop the parens

The new theme includes:

> -   A number of minor improvements to accessibility, including a larger font size, greater use of `aria` attributes, and [accessible syntax highlighting colour scheme](https://apreshill.github.io/rmda11y/arrow.html), designed by Alison Hill.

Reviewer 2: ```suggestion
-   A number of minor improvements to accessibility, including a larger font size, greater use of `aria` attributes, and an [accessible syntax highlighting colour scheme](https://apreshill.github.io/rmda11y/arrow.html), designed by Alison Hill.
```

-   Support for new features like in-line footnotes and [tabsets in articles](https://bookdown.org/yihui/rmarkdown-cookbook/html-tabs.html).

-   Since so many packages have lovely logos, the package logo is now displayed on every page and made even more prominent on the home page.

> As an added incentive to upgrade your template to Bootstrap 5, you'll get site search for "free": pkgdown now supports searching with no external dependencies and no set-up.

Reviewer 2: ```suggestion
As an added incentive to upgrade your template to Bootstrap 5, you'll get site search for "free": pkgdown now supports searching with no external dependencies and no setup.
```

    Activate the translations by setting the `lang` field in `_pkgdown.yaml`, e.g.:

    ``` yaml
    lang: fr
    ```

    pkgdown includes translations for Spanish (es), German (de), French (fr), Portuguese (pt), Turkish (tr) and Chinese (zh_CN).
    A big thanks to my colleagues who provided the initial translations, and to [\@dieghernan](https://github.com/dieghernan), [\@rivaquiroga](https://github.com/rivaquiroga), [\@jplecavalier](https://github.com/jplecavalier) who supplied additional improvements.
    If you're interested in adding translations for your language please [file an issue](https://github.com/r-lib/pkgdown/issues) and we'll help you get started.

-   You can add arbitrary HTML to every page with the new `includes` parameter.
    This makes it easy to add analytics to your site, e.g. to use [plausible.io](https://plausible.io):

    ``` yaml
    templates:
      includes:
        in_header: |
          <script defer data-domain="{YOUR DOMAIN}" src="https://plausible.io/js/plausible.js"></script>
    ```

    Learn more in `?build_site`.

> -   The author, sidebar, and footer configuration is much more flexible allowing you to customise individual components while keeping most of the defaults (previously customisation was mostly all or nothing).

Reviewer 2: ```suggestion
-   The author, sidebar, and footer configuration is much more flexible, allowing you to customise individual components while keeping most of the defaults (previously customisation was mostly all or nothing).
```
